packages/rs-dpp/src/identity/identity_public_key/key_type.rs

```diff
@@ -0,0 +1,94 @@
+use anyhow::bail;
+use bls_signatures::Serialize;
+use ciborium::value::Value as CborValue;
+use dashcore::secp256k1::rand::rngs::StdRng as EcdsaRng;
+use dashcore::secp256k1::rand::SeedableRng;
+use dashcore::secp256k1::Secp256k1;
+use dashcore::Network;
+use itertools::Itertools;
+use lazy_static::lazy_static;
+use rand::rngs::StdRng;
+use rand::Rng;
+use serde_repr::{Deserialize_repr, Serialize_repr};
+use std::collections::HashMap;
+use std::convert::TryFrom;
+
+#[allow(non_camel_case_types)]
+#[repr(u8)]
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Serialize_repr, Deserialize_repr, Hash)]
+pub enum KeyType {
+    ECDSA_SECP256K1 = 0,
+    BLS12_381 = 1,
+    ECDSA_HASH160 = 2,
+    BIP13_SCRIPT_HASH = 3,
+}
+
+lazy_static! {
+    static ref KEY_TYPE_SIZES: HashMap<KeyType, usize> = vec![
+        (KeyType::ECDSA_SECP256K1, 33),
+        (KeyType::BLS12_381, 48),
+        (KeyType::ECDSA_HASH160, 20),
+        (KeyType::BIP13_SCRIPT_HASH, 20)
+    ]
+    .iter()
+    .copied()
+    .collect();
+    pub static ref KEY_TYPE_MAX_SIZE_TYPE: KeyType = KEY_TYPE_SIZES
+        .iter()
+        .sorted_by(|a, b| Ord::cmp(&b.1, &a.1))
+        .last()
+        .map(|(key_type, _)| *key_type)
+        .unwrap();
+}
+
+impl KeyType {
+    /// Gets the default size of the public key
+    pub fn default_size(&self) -> usize {
+        KEY_TYPE_SIZES[self]
+    }
+
+    /// Gets the default size of the public key
+    pub fn random_public_key_data(&self, rng: &mut StdRng) -> Vec<u8> {
```

Comment:
Add a todo to move it under specific feature and module

Reply:
done